util.py

```diff
@@ -80,55 +76,32 @@ def backup_and_rename_temp(filepath, temppath, backuppath=None,
                   (temppath, filepath))
 
 
-def str_to_identifier(string):
-    '''
-    Converts a given string into a usable identifier.
-    Replaces each contiguous sequence of invalid characters(characters
-    unable to be used in a python object name) with a single underscore.
-    If the last character is invalid however, it will be dropped.
-    '''
-    sanitized_str = ''
-    start = 0
-    skipped = False
+non_alphanum_set = r'[^a-zA-Z0-9]+'
+digits_at_start = r'^[0-9]+'
 
-    # make sure the sanitized_strs first character is a valid character
+def str_to_identifier(string):
+    '''Converts given string to a usable identifier. Replaces every sequence
+    of invalid non-alphanumeric characters with an underscore.
+    Trailing underscores are removed.'''
     assert isinstance(string, str)
 
-    while start < len(string):
-        start += 1
-        # ignore characters until an alphabetic one is found
-        if string[start - 1] in ALPHA_IDS:
-            sanitized_str = string[start - 1]
-            break
+    new_string = re.sub(non_alphanum_set, '_', string)
+    new_string = re.sub(digits_at_start, '', new_string)
+    new_string = new_string.rstrip('_')
 
-    # replace all invalid characters with underscores
-    for i in range(start, len(string)):
-        if string[i] in ALPHA_NUMERIC_IDS:
-            sanitized_str += string[i]
-            skipped = False
-        elif not skipped:
-            # no matter how many invalid characters occur in
-            # a row, replace them all with a single underscore
-            sanitized_str += '_'
-            skipped = True
+    assert new_string, "Identifier %s sanitized to an empty string." % (string)

```

Comment:
Wrap the %s in quotes so it is easy to read the original string.

Also, nitpick, "string" being in parentheses does nothing. It isn't being made into a tuple, as a comma is needed to make it one.

Reply:
I'll add the quotes. But I'm keeping the parenthesis for consistency with cases where it needs to be there. Makes the code read more easily for me.